packages/k6-generator/src/index.js

```diff
@@ -0,0 +1,260 @@
+      if (uniquePayload) {
+        const uniqueFile = await new StringPrompt({
+          initial: '../unique/user.json',
+          message: `${methodUrl} request to ${path} requires unique payload - provide a path to JSON file containing this data`,
+        }).run();
+
+        if (
+          !fs.existsSync(`${__dirname}${uniqueFile.slice(uniqueFile.indexOf('/'))}`) ||
+          uniqueFile.slice(-5) !== '.json'
+        ) {
+          console.log(`${uniqueFile} is not a valid path`);
+          process.exit();
+        }
+
+        customImport = `const SLEEP_DURATION = ${methods['sleep']}\nconst uniqueData = JSON.parse(open('${uniqueFile}'))`;
+        uniqueObj = `'${uniqueFile.slice(uniqueFile.lastIndexOf('/') + 1, -5)}'`;
+      }
+      if (!customImport) {
+        customImport = `const SLEEP_DURATION = ${methods['sleep']}`;
+      }
+    }
+  }
+
+  if (!uniqueObj) {
+    uniqueObj = '""';
+  }
+
+  // SPAWN CUSTOM THRESHOLDS AND CHECKS
+  for (let i = 0; i < Object.keys(methods.routes).length; i++) {
+    let methodUrl = Object.keys(methods.routes)[i];
+
+    for (let j = 0; j < methods.routes[methodUrl].length; j++) {
+      const current = methods.routes[methodUrl][j];
+      const { tag } = current;
+      customMetrics = `${customMetrics}
+      "http_req_duration{name:${tag}}": ["p(95)<800"], // threshold on ${tag} requests only
+      "checks{tag:${tag}}": ["rate>0.99"], // rate of successful checks should be higher than 99%`;
+    }
+  }
+
+  // PULL VU OBJECT FROM METHODS FILE - WITHIN DEFAULT FUNCTION
+  const vuObj = Object.assign(methods['vuData']);
+  writeVuObj = `let vuObj = ${JSON.stringify(vuObj)};\n\n`;
+
+  // PARSE THROUGH ALL ROUTES AND ADD TO MAIN CONTENT - WITHIN DEFAULT FUNCTION
+  const baseUrl = methods['baseUrl'];
+
+  for (let i = 0; i < Object.keys(methods.routes).length; i++) {
+    let methodUrl = Object.keys(methods.routes)[i];
+    let fullPath;
+
+    for (let j = 0; j < methods.routes[methodUrl].length; j++) {
+      let current = methods.routes[methodUrl][j];
+      const { path, custom, tag, authReq, payload, uniquePayload, propertyReturned, bodyIncludes } =
+        current;
+      const checkSleep = `check(
+          ${tag}Res, {
+            ['${methodUrl} - ${path} returns successful status']:  (r) => r.status === 200 || r.status === 204,
+            ['${methodUrl} - ${path} returns valid body']: (r) =>
+            r.status === 204 ? true : r.json().hasOwnProperty('${propertyReturned}'),
+            ['${methodUrl} - ${path} returns no error codes']: (r) => r.error_code === 0,
+          },
+          { tag: '${tag}' }
+          );
+          reqBody = {}
+          sleep(SLEEP_DURATION);\n\n`;
+      const setup = `payload = ${JSON.stringify(payload)}
+        if (${payload && payload.length > 0}) {
+            parsePayload(${uniquePayload})
+            }
+        payload = [];
+        tagName = '${tag}Res';`;
+
+      if (custom) {
```

Comment:
What does custom do?

Reply:
The `custom` property indicates if there is something dynamic in the path, for example a `delete` path requiring an `${id}`. I parse the full path differently in this case